pkg/services/validationService.go

```diff
@@ -216,10 +216,46 @@ func (vs *ValidationService) Validate(admissionReviewReq *admission.AdmissionRev
 	return ParseEvaluationResponseIntoAdmissionReview(admissionReviewReq.Request.UID, allowed, msg, *warningMessages), false
 }
 
-type ClusterRequestMetadataAggregator = map[string]*cliClient.ClusterRequestMetadata
+func (m *clusterRequestMetadataMap) Len() int {
+	return m.entriesCount
+}
+
+func (m *clusterRequestMetadataMap) Add(logJson string, clusterRequestMetadata *cliClient.ClusterRequestMetadata) {
+	if m.Load(logJson) == nil {
```

Comment:
A potential bug that most probably won't even happen:
if the value of the key `logJson` is `nil`, the entries count will increase even though it shouldn't.

I believe it can be fixed with:
```golang
if _, ok := m.clusterRequestMetadataAggregator.Load(logJson); ok {
		m.entriesCount += 1
	}
```